src/Uno.UI/DataBinding/BindingPropertyHelper.cs

```diff
@@ -1411,6 +1411,49 @@ private static object UnsetValueGetter(object unused)
 		/// <remarks>This method needs to be aligned with the symbols query in BindableTypeProvidersSourceGenerator.</remarks>
 		private static bool IsValidMetadataProviderType(Type type)
 			=> type.IsPublic && type.IsClass;
+
+		private class GetPropertyTypeKey
+		{
+			private readonly Type _type;
+			private readonly string _property;
+			private readonly bool _allowPrivateMembers;
+
+			private int _cachedHashcode;
+
+			internal GetPropertyTypeKey(Type type, string property, bool allowPrivateMembers)
+			{
+				_type = type;
+				_property = property;
+				_allowPrivateMembers = allowPrivateMembers;
+
+				_cachedHashcode =
+					type.GetHashCode() ^
+					property.GetHashCode() ^
+					allowPrivateMembers.GetHashCode();
+			}
+
+			internal static readonly IEqualityComparer Comparer = new EqualityComparer();
+
+			internal class EqualityComparer : IEqualityComparer
+			{
+				bool IEqualityComparer.Equals(object? x, object? y)
+				{
+					if (x is GetPropertyTypeKey k1 && y is GetPropertyTypeKey k2)
+					{
+						return
+							k1._cachedHashcode == k2._cachedHashcode &&
+							k1._allowPrivateMembers == k2._allowPrivateMembers &&
+							k1._type == k2._type &&
+							k1._property == k2._property;
+
+					}
+
+					return false;
```

Comment:
```suggestion
					return x is null && y is null;
```

Reply:
We won't be inserting null keys so these branches can be avoided.